### DISCUSSION TITLE
Custom port selection for LED DIN

### DISCUSSION THREAD

**Discussion by JuanMeeske** [pixelit-project/PixelIt#370] [Ideas]
First of all, thank you so much for your work on this project. I’ve spent a significant amount of time on it, and I want to take a moment to acknowledge the effort you’ve put in—it truly deserves some kudos.

I’ve played a PCB designer to design a plug-and-play board where you can easily plug & load your pixel clock software. 
![3D_PCB_2024-06-01_S3](https://github.com/user-attachments/assets/47428305-e604-45a5-b6b4-19cbbf5f92da)

I would like to submit the idea to be able to chose the Port for the LED DIN. Currently that is hardcoded to GPIO4/D2.
Similar to how you can currently select the I2C sensor or buttons. This would allow users to choose between ports. In my case GPIO10 or GPIO2 (D4) for the DIN input for the LED matrix.

Building on that, I’d also like to be able to select GPIO4 (D2) as the SDA port for the I2C sensor. 
Currently , that port is being used for the WS2812B.
![image](https://github.com/user-attachments/assets/88e55735-2c40-4851-a4d8-d438dfea9d2e)

**Comment by jj53an** [comment]
I like the idea of having a plug and play board, are you willing to share the PCB design? Saves up some time of soldering. Does it have a I2C sensor on board? GPIO10 is not a valid port on the d1 mini. But I guess that would be the case on your design.

**Comment by foorschtbar** [comment]
Nice design. At the moment, it is no possible to change the matrix pin, sorry.

**Reply by JuanMeeske** [reply]
Took me some time but now I am able to:

1. I cloned this repo: https://github.com/pixelit-project/PixelIt.git
2. change the values in the platformio.ini file 
`-DMATRIX_PIN=D4`
3. Build and upload using PlatformIO

**Comment by m-boy25** [comment]
i am here not for the port but the board i haved the same idea but a larger differant the matrix pre-sodered dht and other captors
soldered a esp8266 soldered is Just a full pixelit but whitout dupont cable
